### ISSUE TITLE
Implement an optional value for the Scylla Helm Chart to add additional rules to ClusterRoleBinding

### ISSUE THREAD

**Issue by niels-s** [scylladb/scylla-operator#440]
* Feature Request

**What should the feature do:**

Implement an optional value as part of the Scylla Helm Chart so we can extend the rules defined for the ClusterRoleBinding generated by the Scylla Helm Chart.

**What is use case behind this feature:**

In our use case, we are running our K8s Cluster using [Kinvolk's Lokomotive](https://github.com/kinvolk/lokomotive) which by default restricts the permissions a ServiceAccount receives when deploying on to the cluster. So we need to add additional rules to the ClusterRoleBinding, otherwise, the Statefulsets generated by the Operator don't have enough permissions to start any pods. We will receive the following error:

```
Warning  FailedCreate      6s (x13 over 27s)  statefulset-controller  create Pod scylla-ewr-1-ewr-1-0 in StatefulSet scylla-ewr-1-ewr-1 failed error: pods "scylla-ewr-1-ewr-1-0" is forbidden: PodSecurityPolicy: unable to admit pod: [spec.initContainers[1].securityContext.privileged: Invalid value: true: Privileged containers are not allowed spec.containers[0].securityContext.capabilities.add: Invalid value: "SYS_NICE": capability may not be added]
```

To resolve this we need to add a rule to give more permission to the service account so it's allowed to add the capabilities. In our case that's:

```
  - apiGroups: [ 'policy' ]
    resources: [ 'podsecuritypolicies' ]
    verbs: [ 'use' ]
    resourceNames:
      - zz-privileged
```

Curious to here your thoughts on this use case

**Comment by tnozicka** [comment]
How about setting up an aggregated ClusterRole? https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles

**Comment by niels-s** [comment]
Interesting, TIL. It looks like an alternative solution. To clarify your comment, do you mean adding an aggregated ClusterRole to the Helm chart in case we need additional rules, or is it your suggestion to manage those outside separately from the Helm chart?

Personally, my preference would go to the former so all the resources needed for a single resource are managed in the same place.

**Comment by tnozicka** [comment]
I was thinking the later, the annotation is added to your custom ClusterRole so the ClusterRole shipped with scylla-operator needs no modification at all. You only create your own role and annotate it so a controller projects it into the scylla-operator ClusterRole.

This is similar to how you'd extend e.g. the default `view`, `edit`, `admin` ClusterRole to allow the regular user to create ScyllaCluster (or any other) CR.

**Comment by niels-s** [comment]
If I understand correctly, your suggestion still requires a change of the helm chart because we would need to update the current ClusterRole to include the aggregation rule to make it work, right?

Something like:

```
aggregationRule:
  clusterRoleSelectors:
  - matchLabels:
      rbac.authorization.k8s.io/aggregate-to-scylla-cluster: "true"
```

I guess the question would be, what's the most straightforward approach. What label would we pick, a generic one or a helm deployment-specific one? If we use a generic one, we lose some flexibility in adding granular permission for a certain deployment.
If we choose a deployment-specific label, I don't see the point of using a workaround using aggregation rules since we could easily add them to the ClusterRole we create, like in #442, without the indirection of the aggregation rule. 

Personally, I prefer having the configuration related to deployment in a single Helm file without having dependencies outside of it to work, which is easier to understand. 

Looking forward to hearing your thoughts on what you had in mind, eventually, there needs to be some mechanism to add more permissions to the ClusterRole. If you can come back to me, I can make a PR with the changes and test it myself since we are setting up several Scylla clusters.

**Comment by tnozicka** [comment]
I was thinking to permanently add the matchLabels part into our ClusterRole, which manifest also into out Helm chart but it will be useful for other deployment types. You'd then just create your "extension" ClusterRole with that label separately.

probably something like `rbac.operator.scylladb.com/aggregate-to-scylla-cluster: "true"`

**Comment by niels-s** [comment]
@tnozicka I've closed the previous PR and created a new one #467 which adds the rules for each helm chart, let me know if this was what you had in mind 😉
